libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -357,7 +357,8 @@ void AP_ICEngine::update(void)
             allow_single_start_while_disarmed = false;
         } else {
             // check if we are blocking disarmed starts
-            if (!allow_single_start_while_disarmed) {
+            if ((hal.util->safety_switch_state() == AP_HAL::Util::SAFETY_DISARMED) ||
```

Comment:
a fairly common setup is to have safety on, but allow control surfaces or throttle though with BRD_SAFETY_MASK, this change looks like it would break that workflow

Reply:
The intention is to leverage the allow a disarmed start MAVLink command for those cases. That is actually how I'm using it.

The problem I'm trying to fix is I managed to start an engine and have 0 control of the throttle, which is a bad situation as well. The surfaces were not enabled, and I had a running engine with no throttle control. We need a solution that protects people in this case. I can add another option bit somewhere, but I think the default expectation would be this.